### PR TITLE
Drawing: Added and set up "Draw" tab in nav bar and some on click handlers.

### PR DESCRIPTION
**Pull request by jmatholia**
Added a tab for the drawing tool. On click handlers, create rectangle on click of canvas and message on click of text.

### CODE REVIEW

File: hs/MakeElements.hs [hunk 1]
```diff
@@ -53,6 +53,13 @@ aboutLinks = concatHtml (map stylesheet ["//netdna.bootstrapcdn.com/bootstrap/3.
                                        "static/style/graph/styles.css",
                                        "static/style/common/about.css",
                                        "static/style/common/common.css"])
+drawLinks :: H.Html
+drawLinks = concatHtml (map stylesheet ["//netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css",
+                                       "static/style/graph/styles.css",
+                                       "static/style/graph/graph_styles.css",
+                                       "static/style/common/about.css",
+                                       "static/style/common/common.css"])
```
Comment (Ian-Stewart-Binks) [inline hs/MakeElements.hs]: Do you need all of these CSS files?

File: hs/DrawResponse.hs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                    theCanvas
+                )
+                timetableScripts
+
+
+drawHtml :: H.Html
+drawHtml = H.div ! A.id "aboutDiv" $ do
+  H.h1 "Draw a Graph"
+  H.p $ do
+    "Here you can create a graph yourself!"
```
Comment (Ian-Stewart-Binks) [inline hs/DrawResponse.hs]: Unless you plan to do more in the `do` block, I don't think that you need to use one here.

File: js/experiment.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+function drawRect() {
+	var c = document.getElementById("theCanvas");
+	var ctx = c.getContext("2d");
+	ctx.fillStyle = "#FF00FF";
```
Comment (Ian-Stewart-Binks) [inline js/experiment.js]: I've created JavaScript code and documentation etiquette guides in the Courseography Wiki in order to make the code share the same features across the code base. These are small things like the use of the single quote `'` over the double quote `"` and JSDocs for function documentation. If you're able to make these changes, I would greatly appreciate it! :smiley_cat:

File: hs/MakeElements.hs (hunk 1 above)
Comment (jmatholia) [inline hs/MakeElements.hs]: Yes, I definitely don't need about.css, and is it just me or styles.css does not exist?

File: hs/DrawResponse.hs (hunk 2 above)
Comment (jmatholia) [inline hs/DrawResponse.hs]: Agreed.

File: js/experiment.js (hunk 3 above)
Comment (jmatholia) [inline js/experiment.js]: Oh yes, definitely.

File: hs/MakeElements.hs (hunk 1 above)
Comment (christinem) [inline hs/MakeElements.hs]: @jmatholia you are right, it doesn't exist anymore, this is a fix I actually made last week and it was merged into happstack by David - you might want to merge happstack with your own branch cause a few fixes have been made to it.

Comment (Ian-Stewart-Binks) [inline hs/MakeElements.hs]: @christinem Merging `happstack` into ones project branch before/while making a pull request makes the pull request diff code very hard to understand. I don't think that this is a good idea.

When merging an issue into the happstack branch, though, merging before is advised.

Comment (christinem) [inline hs/MakeElements.hs]: @Ian-Stewart-Binks mmm, good point, my mistake. I just figured there were a bunch of changes in happstack that had to be at some point updated in our personal branches.

File: hs/MakeElements.hs [hunk 4]
```diff
@@ -43,16 +43,20 @@ timetableLinks = concatHtml (map stylesheet ["//netdna.bootstrapcdn.com/bootstra
...
                                        "static/style/common/common.css"])
 
 aboutLinks :: H.Html
 aboutLinks = concatHtml (map stylesheet ["//netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css",
-                                       "static/style/graph/styles.css",
                                        "static/style/common/about.css",
                                        "static/style/common/common.css"])
+drawLinks :: H.Html
+drawLinks = concatHtml (map stylesheet ["//netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css",
+                                       "static/style/graph/styles.css",
```
Comment (david-yz-liu) [inline hs/MakeElements.hs]: Remove this line

**Comment by david-yz-liu** [comment]
@jmatholia Please do not make any more pulls from happstack. These changes interfere with the review of your pull request.

**Comment by david-yz-liu** [comment]
I think I'll merge this now, but please do address my and Ian's comments for next meeting.

**Pull request merged**